sway-fmt-v2/src/fmt.rs

```diff
@@ -391,6 +401,51 @@ where
 {
     let greeting = 42;
     greeting.to_string()
+}"#;
+        let mut formatter = Formatter::default();
+        let formatted_sway_code =
+            Formatter::format(&mut formatter, Arc::from(sway_code_to_format), None).unwrap();
+        assert_eq!(correct_sway_code, formatted_sway_code)
+    }
+    #[test]
+    fn test_trait_and_super_trait() {
+        let sway_code_to_format = r#"library traits;
+
+trait Person {
+    fn name(self) -> String;
+
+    fn age(self) -> usize;
+}
+
+trait Student: Person {
+    fn university(self) -> String;
+}
+
+trait Programmer {
+    fn fav_language(self) -> String;
+}
+
+trait CompSciStudent: Programmer + Student {
+    fn git_username(self) -> String;
+}"#;
```

Comment:
It looks like there's no difference between the pre-formatted and expected output in this test - should we mess up the `sway_code_to_format` a bit here?

Reply:
Ah yes, I forgot to do that thanks 😅